### PR TITLE
fix: Deploy now working with TypeScript

### PR DESCRIPTION
**Pull request by tbarlow12**
Notes:
- Removed `open` package for now. Contains an invalid typings file. This means service principal authentication is the only way to authenticate
- After the deploy is run, it calls `invoke`, which throws a 404. But that's a separate issue from this one

### CODE REVIEW

File: src/shared/bindings.ts [hunk 1]
```diff
@@ -39,3 +44,11 @@ export function getBindingsMetaData(serverless) {
 
   return parsedBindings;
 }
+
+export async function createEventsBindings(serverless: Serverless, functionName, functionMetadata: IFunctionMetadata): Promise<any> {
```
Comment (mydiemho) [inline src/shared/bindings.ts]: nit: functionName: string

File: src/shared/bindings.ts [hunk 2]
```diff
@@ -39,3 +44,11 @@ export function getBindingsMetaData(serverless) {
...
+  const functionJSON = functionMetadata.params.functionsJson;
+  functionJSON.entryPoint = functionMetadata.entryPoint;
+  functionJSON.scriptFile = functionMetadata.handlerPath;
+  writeFileSync(join(serverless.config.servicePath, functionName + '-function.json'), JSON.stringify(functionJSON, null, 4));
```
Comment (mydiemho) [inline src/shared/bindings.ts]: replace `+` with `,`, and take out `-` in `-function.json`

**Review by mydiemho — APPROVED** [review]
looks good, some nits

File: src/plugins/login/loginPlugin.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export class AzureLoginPlugin {
     this.provider = (this.serverless.getProvider('azure') as any) as AzureProvider;
 
     this.hooks = {
-      'before:deploy:initialize': this.login.bind(this)
+      'package:initialize': this.login.bind(this)
```
Comment (tbarlow12) [inline src/plugins/login/loginPlugin.ts:14]: When using `before:deploy:initialize`, SLS spits out a deprecation warning and instructs to use `package:initialize`

File: src/plugins/login/loginPlugin.ts [hunk 4]
```diff
@@ -28,10 +28,11 @@ export class AzureLoginPlugin {
     try {
       if (subscriptionId && clientId && secret && tenantId) {
         authResult = await loginWithServicePrincipalSecretWithAuthResponse(clientId, secret, tenantId);
-      } else {
-        await open('https://microsoft.com/devicelogin');
-        authResult = await interactiveLoginWithAuthResponse();
-      }
+      }
```
Comment (tbarlow12) [inline src/plugins/login/loginPlugin.ts:31]: Removed `open` for now just because of the invalid typings file

Comment (wbreza) [reply src/plugins/login/loginPlugin.ts:31]: actually - if you install latest typescript it works.

**Pull request merged**